timeinterval/repeating_interval.go

```diff
@@ -0,0 +1,164 @@
+package timeinterval
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+// RepeatingInterval describes an interval with recurring events distributed evenly by a fixed duration.
+// The interval can be bounded by either:
+// a fixed startsAt and endsAt
+// or by a fixed startsAt with a fixed number of Repetitions from which the endsAt will be derived.
+// or by a fixed endsAt with a fixed number of Repetitions from which the startsAt will be derived.
+type RepeatingInterval struct {
+	Interval Interval
+	RepeatIn time.Duration
+	Repetitions *uint32
+}
+
+// UnmarshalJSON unmarshal RepeatingInterval from an ISO8601 "repeating interval" string.
+func (in *RepeatingInterval) UnmarshalJSON(data []byte) error {
+	var s string
+	err := json.Unmarshal(data, &s)
+	if err != nil {
+		return err
+	}
+	ri, err := ParseRepeatingIntervalISO8601(s)
+	if err != nil {
+		return err
+	}
+	*in = *ri
+	return nil
+}
+
+// MarshalJSON marshal RepeatingInterval into an ISO8601 "repeating interval" string.
+func (in RepeatingInterval) MarshalJSON() ([]byte, error) {
+	s, err := in.ISO8601()
+	if err != nil {
+		return nil, err
+	}
+	return json.Marshal(s)
+}
+
+// StartsAt returns the time the interval begins.
+// When possible StartsAt will be derived using the Duration and Repetitions fields if Interval.StartsAt is unset.
+func (in RepeatingInterval) StartsAt() *time.Time {
+	if in.isStartsAtBoundedByRepetitions() {
+		startsAt := in.Interval.EndsAt().Add(-time.Duration(*in.Repetitions)  * in.RepeatIn)
+		return &startsAt
+	}
+	return in.Interval.StartsAt()
+}
+
+// EndsAt returns the time the interval ends.
+// When possible EndsAt will be derived using the Duration and Repetitions fields if Interval.EndsAt is unset.
+func (in RepeatingInterval) EndsAt() *time.Time {
+	if in.isEndsAtBoundedByRepetitions() {
+		endsAt := in.Interval.StartsAt().Add(time.Duration(*in.Repetitions)  * in.RepeatIn)
+		return &endsAt
+	}
+	return in.Interval.EndsAt()
+}
+
+// Duration returns the duration the repeating interval will be active for or nil if it is unbounded.
+func (in RepeatingInterval) Duration() *time.Duration {
+	endsAt := in.EndsAt()
+	startsAt := in.StartsAt()
+	if startsAt == nil || endsAt == nil {
+		return nil
+	}
+	d := endsAt.Sub(*startsAt)
+	return &d
+}
+
+// Started returns a boolean indicating if the interval has begun at the given time.
+func (in RepeatingInterval) Started(t time.Time) bool {
+	if in.isStartsAtBoundedByRepetitions() {
+		startsAt := in.StartsAt()
+		return t.Equal(*startsAt) || t.After(*startsAt)
+	}
+	return in.Interval.Started(t)
+}
+
+// Ended returns a boolean indicating if the interval has ended at the given time.
```

Comment:
Same for this. Wouldn't it return true for everything in between beginning and end?